documentation/bolt_examples.md

```diff
@@ -123,3 +123,226 @@ To convert the `restart_service.ps1` script to a task, giving you the ability to
+
+```     
+# Modules from the Puppet Forge
+mod 'puppetlabs-chocolatey', '4.1.0'
+## dependencies
+mod 'puppetlabs-stdlib', '4.13.1' #install latest
+ mod 'puppetlabs-powershell', '2.3.0'
+mod 'puppetlabs-registry', '2.1.0'
+        
+# Modules from Git
+# Examples: https://github.com/puppetlabs/r10k/blob/master/doc/puppetfile.mkd#examples
+#mod 'apache',
+#  :git    => 'https://github.com/puppetlabs/puppetlabs-apache',
+#  :commit => 'de290646f97e04b4b8e42c70f6e01e860c394ce7'     
+ ```
+
+2. Add the Puppetfile to a version control service, for example, Github.
+
+3. Add the following code to the Puppetfile — with your git user account: 
+
+```     
+mod 'puppet_choco_tap',
+    :git    =>  'https://github.com/<your git user>/puppet_choco_tap.git'
+```
+
+4. From inside the project directory, install the Forge content by running: 
+
+```
+bolt puppetfile install
+```
+
+After it runs, you can see a `modules` directory inside the project directory, containing the modules you specified in the `Puppetfile`.
+
+### 3. Write a Bolt plan to apply Puppet code and orchestrate the deployment of a package resource using the Chocolatey provider. Plans allow you to run more than one task with a single command.
+
+1. Inside your project directory, in a file called `/plan/installer.pp`, create a plan called `puppet_chocolatey_tap::installer`. 
+
+The folder tree should look like this:
+
+```
+├──plan
+   └── installer.pp 
+```
+
+2. Copy the following code into the `installer.pp` file:
+
+```
+plan puppet_choco_tap::installer(
+   TargetSpec                                   $nodes,
+   String                                       $package,
+   Variant[Enum['absent', 'present'], String ]  $ensure = 'present',
+ ){
+
+ apply_prep($nodes)
+
+apply($nodes){
+  include chocolatey
+ }
+apply($nodes){
+  package { $package :
+    ensure    => $ensure,
+    provider  => 'chocolatey',
+    }
+  }
+}
+```
+
+Take note of the following features of the plan:
+
+- It has three parameters: the `TargetSpec`, a `package` string for the package name, and the `ensure` state of the package which allows for version, absent or present.
+- It has the `apply_prep` function call, which is used to install modules needed by `apply` on remote nodes as well as to gather facts about the nodes.
+- The first `apply` block installs the Chocolatey package manager. The Chocolatey provider is also deployed as a library with the Puppet agent in `apply_prep`.
+- The second `apply` block installs a specified package using the Chocolatey provider.
+
+2. To verify that the `puppet_chocolatey_tap` plan is available, run the following command: 
+
+```
+bolt plan show
+```
+
+The output looks like this:
+
+```
+facts
+facts::info
+puppet_choco_tap::installer
+puppetdb_fact
+```
+
+3. Run the plan with the `bolt plan run` command: 
+
+```
+bolt plan run puppet_choco_tap::installer package=frogsay --nodes=win --password
```

Comment:
Wrong target for inventory above. Should be windows, also do not prompt for password bc it is in the inventoryfile.

Reply:
Also, nit, but use `--targets` not `--nodes`.